tools/api-performance-test/performance.py

```diff
@@ -0,0 +1,333 @@
+  ecosystem = random.choice(bug_map[request_id].ecosystems)
+  return {
+      "version": bug_map[request_id].affected_fuzzy,
+      "package": {
+          "name": package,
+          "ecosystem": ecosystem
+      }
+  }
+
+
+def batch_payload(request_ids, bug_map):
+  """batch query payload"""
+  size = random.randint(1, 100)
+  batch_ids = random.sample(request_ids, min(size, len(request_ids)))
+  queries = []
+  for bug_id in batch_ids:
+    query = {}
+    query_type = random.choice(['version', 'project'])
+    if query_type == 'version':
+      query = version_payload(bug_id, bug_map)
+    elif query_type == 'project':
+      query = package_payload(bug_id, bug_map)
+    queries.append(query)
+
+  return {"queries": [queries]}
+
+
+def vulnerability_payload():
+  """vulnerability query doesn't need request body"""
+  return None
+
+
+def get_large_batch_query(project_map):
+  """Gets packages with the most amount of vulns."""
+  most_common = 5000
+  project_counter = Counter()
+  for project in project_map:
+    # filter out invalid project name and Linux Kernel
+    if project in ('foo', 'Kernel'):
+      continue
+    project_counter[project] = len(project_map[project])
+  most_vulnerable_projects = project_counter.most_common(most_common)
+  large_batch_query_ids = []
+  for project, project_count in most_vulnerable_projects:
+    if project_count == 10:
+      break
+    large_batch_query_ids.append(project_map[project].pop())
+
+  random.shuffle(large_batch_query_ids)
+  return large_batch_query_ids
+
+
+async def send_version_requests(request_ids, bug_map):
+  """Sends version query requests"""
+  url = f'{BASE_URL}/query'
+  batch_size = VERSION_QUERY_BATCH_SIZE
+  await make_http_requests_async(request_ids, bug_map, url, batch_size,
+                                 version_payload)
+
+
+async def send_package_requests(request_ids, bug_map):
+  """Sends package query requests"""
+  url = f'{BASE_URL}/query'
+  batch_size = PACKAGE_QUERY_BATCH_SIZE
+  await make_http_requests_async(request_ids, bug_map, url, batch_size,
+                                 package_payload)
+
+
+async def send_vuln_requests(request_ids, bug_map):
+  """Sends vulnerability get requests"""
+  url = f'{BASE_URL}/vulns'
+  batch_size = VULN_QUERY_BATCH_SIZE
+  await make_http_requests_async(request_ids, bug_map, url, batch_size,
+                                 vulnerability_payload)
+
+
+async def send_batch_requests(request_ids, bug_map, batch_size):
+  """Sends batch query requests"""
+  url = f'{BASE_URL}/querybatch'
+  await make_http_requests_async(request_ids, bug_map, url, batch_size,
+                                 batch_payload)
+
+
+async def main():
+  """Main"""
+  if not os.path.exists(BUG_DIR):
+    # This will take around 10 mins
+    get_bugs_from_datastore()
+
+  # The `ecosystem_map` can be used to filter our queries for a
+  # specific ecosystem.
+  ecosystem_map, bug_map, project_map = load_all_bugs(BUG_DIR)
```

Comment:
I know it's called 'project' in the datastore, but I think it'll be clearer if we name this package instead of project (project makes me think of the gcp project.)